catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v110/catalog/source/WfsSourceTest.java

```diff
@@ -1371,6 +1389,210 @@ public void testQuerySendsHitsRequestBeforeResultsRequest() throws Exception {
+
+    mapSchemaToFeatures(ONE_TEXT_PROPERTY_SCHEMA_PERSON, ONE_FEATURE);
+    setUpMocks(null, null, ONE_FEATURE, ONE_FEATURE);
+    final QueryImpl propertyIsLikeQuery =
+        new QueryImpl(builder.attribute(Metacard.ANY_TEXT).is().like().text("literal"));
+    setupMapper(
+        MOCK_TEMPORAL_SORT_PROEPRTY, MOCK_RELEVANCE_SORT_PROEPRTY, MOCK_DISTANCE_SORT_PROEPRTY);
+    source.setMetacardMappers(metacardMappers);
+    propertyIsLikeQuery.setSortBy(new SortByImpl(Result.TEMPORAL, (String) null));
+
+    final ArgumentCaptor<GetFeatureType> captor = ArgumentCaptor.forClass(GetFeatureType.class);
+    source.query(new QueryRequestImpl(propertyIsLikeQuery));
+  }
+
+  @Test
+  public void testSortingNoSortProperty() throws Exception {
+    // if sort property is missing, throw UnsupportedQueryException
+    expectedEx.expect(UnsupportedQueryException.class);
+    expectedEx.expectMessage(
+        "Source WFS_ID does not support specified sort property null with sort order SortOrder[ASCENDING]");
+
+    mapSchemaToFeatures(ONE_TEXT_PROPERTY_SCHEMA_PERSON, ONE_FEATURE);
+    setUpMocks(null, null, ONE_FEATURE, ONE_FEATURE);
+    final QueryImpl propertyIsLikeQuery =
+        new QueryImpl(builder.attribute(Metacard.ANY_TEXT).is().like().text("literal"));
+    setupMapper(
+        MOCK_TEMPORAL_SORT_PROEPRTY, MOCK_RELEVANCE_SORT_PROEPRTY, MOCK_DISTANCE_SORT_PROEPRTY);
+    source.setMetacardMappers(metacardMappers);
+    propertyIsLikeQuery.setSortBy(new SortByImpl(null, "ASC"));
+
+    final ArgumentCaptor<GetFeatureType> captor = ArgumentCaptor.forClass(GetFeatureType.class);
+    source.query(new QueryRequestImpl(propertyIsLikeQuery));
+  }
+
+  @Test
+  public void testSortingBadSortOrder() throws Exception {
+    // if sort order is invalid throw UnsupportedQueryException
+    expectedEx.expect(UnsupportedQueryException.class);
+    expectedEx.expectMessage(
+        "Source WFS_ID does not support specified sort property TEMPORAL with sort order SortOrder[foo]");
+
+    mapSchemaToFeatures(ONE_TEXT_PROPERTY_SCHEMA_PERSON, ONE_FEATURE);
+    setUpMocks(null, null, ONE_FEATURE, ONE_FEATURE);
+    final QueryImpl propertyIsLikeQuery =
+        new QueryImpl(builder.attribute(Metacard.ANY_TEXT).is().like().text("literal"));
+    setupMapper(
+        MOCK_TEMPORAL_SORT_PROEPRTY, MOCK_RELEVANCE_SORT_PROEPRTY, MOCK_DISTANCE_SORT_PROEPRTY);
+    source.setMetacardMappers(metacardMappers);
+    propertyIsLikeQuery.setSortBy(new SortByImpl(Result.TEMPORAL, "foo"));
+
+    final ArgumentCaptor<GetFeatureType> captor = ArgumentCaptor.forClass(GetFeatureType.class);
+    source.query(new QueryRequestImpl(propertyIsLikeQuery));
+  }
+
+  private void assertFeature(
+      GetFeatureType getFeatureType,
+      boolean sortingEnabled,
+      String sortProperty,
+      String sortOrder) {
+    assertThat(getFeatureType.getQuery().size(), is(ONE_FEATURE));
```

Comment:
✏️ Assertions on collection size should use the [`hasSize()`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#hasSize(int)) matcher for more useful messages when the test fails.